C23-incompatible bool definition and CMake build

### What type of issue is this?

permanent - occurring repeatedly

### What issue are you facing?

The build system used to explicitly set the C standard to C90 with `set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu90")` - this was removed in the #1586 changes. This causes build failure with a compiler like GCC 15 that defaults to C23, because several of the source files have non-C23-compatible definitions of `bool`.

There are two ways to fix this: either specify the C standard explicitly in CMake (it's simpler to do this by setting `CMAKE_C_STANDARD`), or fix the code to be compatible with newer versions of C (you could require C99 rather than C90 and use `stdbool.h`, for example).

### What are the steps to reproduce this?

Build using CMake's defaults on a machine with GCC 15.

### Can you provide any logs? (output, errors, etc.)

You get an error like this for each of the files that has a `typedef` for `bool`:
```
/src/radio/hackrf/work/ab605692553a9e59dd74222f98fa1dda86a5e44e/host/hackrf-tools/src/hackrf_operacake.c:31:13: error: ‘bool’ cannot be defined via ‘typedef’
   31 | typedef int bool;
      |             ^~~~
/src/radio/hackrf/work/ab605692553a9e59dd74222f98fa1dda86a5e44e/host/hackrf-tools/src/hackrf_operacake.c:31:13: note: ‘bool’ is a keyword with ‘-std=c23’ onwards
/src/radio/hackrf/work/ab605692553a9e59dd74222f98fa1dda86a5e44e/host/hackrf-tools/src/hackrf_operacake.c:31:1: warning: useless type name in empty declaration
   31 | typedef int bool;
      | ^~~~~~~
```

--- a/host/hackrf-tools/src/hackrf_clock.c
+++ b/host/hackrf-tools/src/hackrf_clock.c
@@ -22,16 +22,11 @@
 
 #include <hackrf.h>
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 #include <getopt.h>
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #define CLOCK_UNDEFINED  0xFF
 #define REGISTER_INVALID 32767

--- a/host/hackrf-tools/src/hackrf_debug.c
+++ b/host/hackrf-tools/src/hackrf_debug.c
@@ -24,16 +24,11 @@
 
 #include <hackrf.h>
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 #include <getopt.h>
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #define REGISTER_INVALID 32767
 

--- a/host/hackrf-tools/src/hackrf_operacake.c
+++ b/host/hackrf-tools/src/hackrf_operacake.c
@@ -22,16 +22,11 @@
 
 #include <hackrf.h>
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <getopt.h>
 #include <string.h>
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #define FREQ_MIN_MHZ (0)    /*    0 MHz */
 #define FREQ_MAX_MHZ (7250) /* 7250 MHz */

--- a/host/hackrf-tools/src/hackrf_spiflash.c
+++ b/host/hackrf-tools/src/hackrf_spiflash.c
@@ -23,17 +23,12 @@
 
 #include <hackrf.h>
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <getopt.h>
 #include <sys/types.h>
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #ifdef _MSC_VER
 	#ifdef _WIN64

--- a/host/hackrf-tools/src/hackrf_sweep.c
+++ b/host/hackrf-tools/src/hackrf_sweep.c
@@ -23,6 +23,7 @@
 
 #include <hackrf.h>
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -37,12 +38,6 @@
 #include <inttypes.h>
 
 #define _FILE_OFFSET_BITS 64
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #ifdef _WIN32
 	#define _USE_MATH_DEFINES

--- a/host/hackrf-tools/src/hackrf_transfer.c
+++ b/host/hackrf-tools/src/hackrf_transfer.c
@@ -25,6 +25,7 @@
 
 #include <hackrf.h>
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -37,12 +38,6 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <inttypes.h>
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #ifdef _WIN32
 	#include <windows.h>

--- a/host/libhackrf/src/hackrf.c
+++ b/host/libhackrf/src/hackrf.c
@@ -23,6 +23,7 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
 
 #include "hackrf.h"
 
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #ifndef _WIN32
@@ -37,12 +38,6 @@ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSI
 	#define strdup               _strdup
 #endif
 #include <pthread.h>
-
-#ifndef bool
-typedef int bool;
-	#define true 1
-	#define false 0
-#endif
 
 #ifdef HACKRF_BIG_ENDIAN
 	#define TO_LE(x)     __builtin_bswap32(x)

